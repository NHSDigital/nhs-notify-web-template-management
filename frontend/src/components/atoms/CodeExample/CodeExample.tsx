import React, { HTMLProps } from 'react';

interface CodeExampleProps extends HTMLProps<HTMLDivElement> {
  ariaText: string;
  ariaId: string;
  codeClassName?: string;
}

const CodeExample: React.FC<CodeExampleProps> = ({
  ariaText,
  ariaId,
  children,
  codeClassName,
}) => {
  return (
    <>
      <span id={ariaId} className='nhsuk-u-visually-hidden'>
        {ariaText}
      </span>
      <pre className='nhsuk-code-block'>
        <code aria-describedby={ariaId} className={codeClassName}>
          {children}
        </code>
      </pre>
    </>
  );
};

export default CodeExample;
