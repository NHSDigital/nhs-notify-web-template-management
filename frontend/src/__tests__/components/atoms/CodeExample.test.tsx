import { render } from '@testing-library/react';
import CodeExample from '@atoms/CodeExample/CodeExample';

describe('CodeExample component', () => {
  it('renders component correctly', () => {
    const container = render(
      <CodeExample ariaId='test-id' ariaText='This is an example of markdown:'>
        # A test example
      </CodeExample>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('renders component correctly when optional class is provided', () => {
    const container = render(
      <CodeExample
        ariaId='test-id'
        ariaText='This is an example of markdown:'
        codeClassName='test-class'
      >
        # A test example
      </CodeExample>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
