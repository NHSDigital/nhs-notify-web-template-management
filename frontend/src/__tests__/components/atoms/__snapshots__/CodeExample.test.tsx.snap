// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeExample component renders component correctly 1`] = `
<DocumentFragment>
  <span
    class="nhsuk-u-visually-hidden"
    id="test-id"
  >
    This is an example of markdown:
  </span>
  <code
    aria-describedby="test-id"
  >
    # A test example
  </code>
</DocumentFragment>
`;

exports[`CodeExample component renders component correctly when optional class is provided 1`] = `
<DocumentFragment>
  <span
    class="nhsuk-u-visually-hidden"
    id="test-id"
  >
    This is an example of markdown:
  </span>
  <code
    aria-describedby="test-id"
    class="test-class"
  >
    # A test example
  </code>
</DocumentFragment>
`;
